flink-python/pyflink/datastream/functions.py

```diff
@@ -119,6 +119,40 @@ def get_aggregating_state(
         pass
 
 
+class OperatorStateStore(ABC):
+    """
+    Interface for getting operator states. Currently, only :class:`~state.BroadcastState` is
+    supported.
+    .. versionadded:: 1.16.0
+    """
+
+    @abstractmethod
+    def get_broadcast_state(self, state_descriptor: MapStateDescriptor) -> BroadcastState:
+        """
+        Fetches the :class:`~state.BroadcastState` described by :class:`~state.MapStateDescriptor`,
+        which has read/write access to the broadcast operator state.
+        """
+        pass
+
+    @abstractmethod
+    def get_read_only_broadcast_state(
```

Comment:
I don't think `OperatorStateStore` should have this method. Whether `BroadcastState` is readonly should depend on whether it is in `process_element` or `process_broad_cast_element`